src/dmd/aliasthis.d

```diff
@@ -181,3 +181,45 @@ bool checkDeprecatedAliasThis(AliasThis at, const ref Loc loc, Scope* sc)
     }
     return false;
 }
+
+/**************************************
+ * Check and set 'att' if 't' is a recursive 'alias this' type
+ * Params:
+ *   att = type reference used to detect recursion
+ *   t   = 'alias this' type
+ *
+ * Returns:
+ *   Whether the 'alias this' is recursive or not
+ */
+bool isRecursiveAliasThis(ref Type att, Type t)
+{
+    auto tb = t.toBasetype();
+    if (att && tb.equivalent(att))
+        return true;
+    else if (!att && tb.checkAliasThisRec())
+        att = tb;
+    return false;
+}
+
+/**************************************
+ * Check and set 'e.attX' if 'e' has recursive 'alias this' type in 'e1' or 'e2'
+ * Params:
+ *   side = "e1" for e.e1 or "e2" for e.e2
+ *   e    = expression with 'alias this' in e1 or e2
+ *
+ * Returns:
+ *   Whether the 'alias this' is recursive or not
+ */
+bool isRecursiveAliasThis(string side, E)(E e)
```

Comment:
If I'm not missing something obvious here, I think that this template is overkill. Instead of:

```d
isRecursiveAliasThis!"e1"(ae)`
```
One could simply call
```d
isRecursiveAliasThis(ae.att1, ae.e1.type)
```

And we don't get all the extra-compile time associated with templates and object file size increase.

Reply:
Yes, I don't remember exactly the reason of this templated version I supposed it was my first function that was forked to handle other cases and to avoid a possible input mistake between `e.att1/2` and `e.e1/2`.

Reply:
Template removed.